src/_guides/libraries/java-interop.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Java interop using package:jnigen"
+description: "To use Java in your Dart program, use package:jnigen."
+jnigen: "https://pub.dev/packages/jnigen"
+jni: "https://pub.dev/packages/jni"
+example: "https://github.com/HosseinYousefi/jnigen_example/tree/main"
+jnidoc: "https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/jniTOC.html"
+---
+
+Dart mobile, command-line, and server apps
+running on the [Dart Native platform](/overview#platform), on Android, Windows
+and Linux can use [`package:jni`]({{page.jni}})
+and [`package:jnigen`]({{page.jnigen}})
+to call Java and Kotlin APIs.
+
+{{site.alert.note}}
+  This interop feature is **experimental**,
+  and [in active development](https://github.com/dart-lang/jnigen/issues).
+{{site.alert.end}}
+
+`package:jni` allows Dart code to interact
+with Java through [JNI]({{page.jnidoc}}). 
+However, doing so involves a lot of boilerplate code,
+so you can use `package:jnigen` to automatically generate
+the Dart bindings for a given Java API.
+
+You can decompile Kotlin to Java, allowing `package:jnigen`
+to generate bindings for Kotlin as well.
+
+## Simple Java Example
+
+This guide walks you through [an example]({{page.example}})
+that uses `package:jnigen` to generate bindings for a simple class
+that can show a toast.
+
+Along with JDK, maven is required. 
+
+It's recommended to have `clang-format` installed to format the generated
+C bindings.
+
+### Configuring jnigen
+
+First, add `package:jni` as a dependency and `package:jnigen`
+as a dev dependency.
+
+```terminal
+$ dart pub add jni
+$ dart pub add --dev jnigen
```

Comment:
`pub add` has [new recommended syntax](https://dart.dev/tools/pub/cmd/pub-add#dev-dependency)! So at minimum this needs to be written like:
 ```suggestion
$ dart pub add jni
$ dart pub add dev:jnigen
```

But you could also add them in the same line:
```suggestion
$ dart pub add jni dev:jnigen
```